validate.go

```diff
@@ -140,8 +145,13 @@ func (ctx *ValidationContext) verifySignedInfo(signatureElement *etree.Element,
 	}
 
 	// Canonicalize the xml
+	canonical, err := canonicalize(signedInfo, AlgorithmID(c14nAlgorithmId))
+	if err != nil {
+		return err
+	}
+
 	doc := etree.NewDocument()
-	doc.SetRoot(canonicalize(signedInfo, SignatureAlgorithm(signatureMethodId)))
```

Comment:
This was the underlying bug - `signatureMethodId` was clearly wrong here. It was also odd that `canonicalize` took a `SignatureAlgorithm` as a parameter, which led me to clean up how algorithm IDs are typed.